File: 00881.py
class Solution:
    def numRescueBoats(self, people: list[int], limit: int) -> int:
        people.sort()
        start = 0; end = len(people)-1
        cnt = 0
        while start <= end:
            if people[start]+people[end] <= limit:
                start += 1
                end -= 1
                cnt += 1
            else:
                end -= 1
                cnt += 1
        return cnt

File: ProblemList/00881.cpp
#include <vector>
using namespace std;
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(), people.end());
        int ans = 0, left = 0, right = people.size()-1;
        while(left <= right){
            if(people[left] + people[right] > limit) right -= 1;
            else{
                left += 1;
                right -= 1;
            }
                ans += 1;
        }
        return ans;
    }
};